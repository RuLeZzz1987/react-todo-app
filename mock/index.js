export default {
  categories: [
    {name: 'Category_1', id: 5},
    {name: 'Category_2', id: 1},
    {name: 'Category_1_1', id: 2, parentId: 5},
    {name: 'Category_1_1_1', id: 3, parentId: 2},
    {name: 'Category_2_1', id: 4, parentId: 1}
  ],

  todos: [
    {name: 'Todo_5', categoryId: 2, id: 5},
    {name: 'Todo_1', categoryId: 1, id: 1},
    {name: 'Todo_2', categoryId: 3, id: 2},
    {name: 'Todo_3', categoryId: 4, id: 3},
    {name: 'Todo_4', categoryId: 4, id: 4}
  ]
};