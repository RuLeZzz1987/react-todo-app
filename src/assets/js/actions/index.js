export CategoryActions from './CategoryActions';
export TodoActions from './TodoActions';
export FilterActions from './FilterActions';