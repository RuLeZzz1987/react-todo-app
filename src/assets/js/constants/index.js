
export const CATEGORY = Symbol('category');

export const TODO = Symbol('todo');