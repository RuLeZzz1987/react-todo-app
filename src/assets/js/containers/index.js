export Header from './HeaderContainer';
export Categories from './CategoriesContainer';
export Category from './CategoryContainer';
export Editor from './EditorContainer';
export ModalError from './ModalErrorContainer';