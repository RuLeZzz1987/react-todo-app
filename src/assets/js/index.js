import React from 'react';
import { render } from 'react-dom';
import App from './components'
import '../styles/styles.scss';

render(<App/>, document.querySelector('#root'));