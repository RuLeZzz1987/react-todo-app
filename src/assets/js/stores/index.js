export TodoStore from './TodoStore';
export CategoryStore from './CategoryStore';
export FilterStore from './FilterStore';
export ErrorStore from './ErrorStore';