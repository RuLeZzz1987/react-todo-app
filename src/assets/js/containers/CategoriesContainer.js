import React, { Component } from 'react';
import { Container } from 'flux/utils'
import { CategoryStore, FilterStore, TodoStore } from '../stores';
import Categories from '../components/categories';
import { CategoryActions } from '../actions';
import { reduceWithCondition } from '../helpers';

class CategoriesContainer extends Component {

  static getStores() {
    return [CategoryStore, FilterStore, TodoStore]
  }

  static calculateState(prevState, props) {
    const { categoryId, todoId } = props.params;

    const todos = TodoStore.getState();
    const filters = FilterStore.getState();
    let ids = Object.keys(
      reduceWithCondition(
        reduceWithCondition(CategoryStore.getState())('parentId', undefined)
      )('isComplete', filters.showDone));

    return {
      ids,
      selectedId: props.categoryId,
      isTodoFound: Object.keys(todos).some(id => id == todoId && todos[id].parentId == categoryId)
    }
  }

  render() {
    return (
      <Categories
        {...this.state}
        {...this.props}
        addCategory={CategoryActions.addCategory}
      />
    )
  }
}

export default Container.create(CategoriesContainer, {withProps: true})