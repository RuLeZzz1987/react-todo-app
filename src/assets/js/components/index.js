import React, { Component } from "react";
import { Header } from "../containers";
import Main from "./main";
import { Category, Todo } from "../models";

const defaultCategories = [new Category({name: 'Category_1', isRoot: true, children: [new Todo({name: 'To-Do Item #1'}), new Category({name: 'Category_1_1'})]})];

class App extends Component {
    
    constructor(props) {
        super(props);
        
        this.state = {
            categories: defaultCategories,
            showDone: true,
            filteredCategories: defaultCategories
        };
        
        this.updateCategories = (categories, cb) => this.setState({categories, filteredCategories: categories}, cb);
    }
    
    render() {
        return (
            <div>
                <Header
                    todoId={this.props.params.todoId}
                />
                <Main
                    params={this.props.params}
                />
            </div>
        )
    }
}

export default App;