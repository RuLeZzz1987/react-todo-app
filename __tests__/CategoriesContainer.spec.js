import React from 'react';
import renderer from 'react-test-renderer';
import { Categories } from '../src/assets/js/containers';
import * as Stores from '../src/assets/js/stores';
import mock from '../mock';

describe('CategoriesContainer', function () {

  beforeEach(function () {
    Object.keys(Stores).forEach(key=>{
      Stores[key]._state = Stores[key].getInitialState();
    });

    this.render = (params={}) => renderer.create(<Categories params={params}/>).toJSON();

  });

  it('can render empty Categories block', function () {

    expect(this.render()).toMatchSnapshot();

  });

  it('can render Categories block showing only not completed items', function () {

    Stores.FilterStore._state.showDone = false;

    expect(this.render()).toMatchSnapshot();

  });

  it('can render Categories tree', function () {
    Stores.CategoryStore._state = mock.categoryStore;
    Stores.TodoStore._state = mock.todoStore;

    const params = {
      categoryId: '1',
      todoId: '1'
    };

    expect(this.render(params)).toMatchSnapshot();
  });

});