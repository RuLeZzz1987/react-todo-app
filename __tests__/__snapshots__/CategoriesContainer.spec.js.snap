exports[`CategoriesContainer can render Categories tree 1`] = `
<aside
  className="categories">
  <section
    className="editor-area">
    <div
      className="editor">
      <input
        name="category"
        onChange={[Function]}
        placeholder="Enter category title"
        type="text"
        value="" />
      <div
        className="add-button"
        onClick={[Function]}>
        <span
          className="add-label">
          ADD
        </span>
      </div>
    </div>
  </section>
  <section
    className="categories-tree" />
</aside>
`;

exports[`CategoriesContainer can render empty Categories block 1`] = `
<aside
  className="categories">
  <section
    className="editor-area">
    <div
      className="editor">
      <input
        name="category"
        onChange={[Function]}
        placeholder="Enter category title"
        type="text"
        value="" />
      <div
        className="add-button"
        onClick={[Function]}>
        <span
          className="add-label">
          ADD
        </span>
      </div>
    </div>
  </section>
  <section
    className="categories-tree" />
</aside>
`;
